### DISCUSSION TITLE
Customize default focus styles for tailwindcss/forms

### DISCUSSION THREAD

**Discussion by helhum** [tailwindlabs/tailwindcss#4645] [Help] [answered]
I have configured a custom ring color in tailwindcss theme config like so:

```js
    theme: {
        extend: {
            ringColor: (theme) => ({
                DEFAULT: theme('colors.grey.500'),
            }),
```

and apply these to input fields like this:

```html
<input type="text" class="focus:outline-none focus:ring" placeholder="Placeholder" />
```

When I now activate the tailwindcss/forms plugin, this input field gets the default focus ring color delivered with this plugin and I could not find a way to customize this color.

Ideally I'd like to have the forms plugin respect the ring color I configured in the theming config.

Any way to achieve this?

**Comment by VicGUTT** [comment, marked as answer]
Hey !

Taking a quick look at the [@tailwindcss/forms plugin's source code](https://github.com/tailwindlabs/tailwindcss-forms/blob/6f951b232c23ef8766a56d624a312ce6d72f6c53/src/index.js#L49), the default ring color is set there, and in a none customizable manner.

So the only solutions I see is to either propose a pull request to change `'--tw-ring-color': theme('colors.blue.600', colors.blue[600])` to something like `'--tw-ring-color': theme('ringColor.DEFAULT', theme('colors.blue.600', colors.blue[600])` or maybe to propose allowing customizing the plugin's styles entirely, like:

```js
// tailwind.config.js
theme: {
    forms: {
        '[type="text"], [type="email"], ...': (theme) => ({
            //
        }),
    },
},

// Plugin's source code
addBase(
    rules
        .map((rule) => {
            if (rule[strategy] === null) {
                return null
            }

            return { [rule[strategy]]: theme(`forms.${rule[strategy]}`, rule.styles) }
        })
        .filter(Boolean)
)
```

Hope it help :v:

**Reply by helhum** [reply]
Thanks for the confirmation, that customisation currently isn't possible. I created a feature request here: https://github.com/tailwindlabs/tailwindcss/discussions/4649
